tee-worker/litentry/core/stf-task/receiver/src/handler/assertion.rs

```diff
@@ -65,17 +101,31 @@ where
 	H::StateT: SgxExternalitiesTrait,
 {
 	type Error = VCMPError;
-	type Result = ();
+	type Result = Option<(Vec<u8>, Vec<u8>)>;
 
 	fn on_process(&self) -> Result<Self::Result, Self::Error> {
 		match self.req.assertion.clone() {
-			Assertion::A1 => lc_assertion_build::a1::build(self.req.vec_identity.to_vec()),
+			Assertion::A1 => lc_assertion_build::a1::build(
+				self.req.vec_identity.clone(),
+				self.req.credential.clone(),
+			)
+			.map(|credential| {
+				let encoded_callback = TrustedCallFork::build_assertion_runtime(
+					self.context.enclave_signer.get_enclave_account().unwrap(),
+					self.req.who.clone(),
+					Box::new(credential),
```

Comment:
A few things that I don't understand:
- why do we need `TrustedCallFork`? If you need to change something from the `encoded_callback`, you could decode it, extract any necessary parameters and rebuild the encoded callback.
- this would call `build_assertion_runtime` synchronously - but it's another thread, we don't have any sync mechanism to shared data (e.g. sp_io storages). I think it's best to submit the callback back to the top pool. 
- maybe we don't need to submit it back to the top pool at all. Can't we simply send the result to parachain? If you need the shielding key you could simply pass it in as request parameters.

Please feel free to share your thoughts too @zhizming-zhong

Reply:
`TrustedCallFork` is kind of a hack to the `encoded_callback`.

Generating a credential needs the following steps: 
1. Generating an unsigned credential from templates  <-- `TrustedCallSigned build_assertion_preflight`
2. Check the assertion and get its built return value  <-- `lc-assertion-build` processed by stf_task receiver
3. Add Issuer's enclave metadata (more might be added if needed) <-- `TrustedCallSigned build_assertion_runtime`
4. Sign the credential by enclave <-- `TrustedCallSigned build_assertion_runtime`
5. Pass the signed credential back to parachain <-- `TrustedCallSigned build_assertion_runtime`

The major difficulty is `unsigned_credential` is needed to be shared between stf-task sender and receiver, but there isn't a sync mechanism or a runtime pallet for it to share/store the value.

 Let's discuss it offline to figure out how to make the workflow more simple and clear.

Reply:
I think the biggest problem is that crate `stf` can't be imported into `stf_task_receiver`, so that there will be a circular dependency problem